configs/example/gem5_library/checkpoints/simpoints-se-restore.py

```diff
@@ -0,0 +1,70 @@
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+from gem5.components.cachehierarchies.classic.\
+    private_l1_private_l2_cache_hierarchy import(
+    PrivateL1PrivateL2CacheHierarchy
+)
+from gem5.components.boards.simple_board import SimpleBoard
+from gem5.components.memory import DualChannelDDR4_2400
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.isas import ISA
+from gem5.resources.resource import Resource
+
+from gem5.components.processors.simpoint import SimPoint
+from pathlib import Path
+
+cache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
+    l1d_size = "32kB",
+    l1i_size="32kB",
+    l2_size="256kB",
+)
+
+requires(isa_required=ISA.X86)
+
+memory = DualChannelDDR4_2400(size = "3GB")
+
+processor = SimpleProcessor(
+    cpu_type=CPUTypes.TIMING,
+    isa=ISA.X86,
+    num_cores=1,
+)
+
+board = SimpleBoard(
+    clk_freq="3GHz",
+    processor=processor,
+    memory=memory,
+    cache_hierarchy=cache_hierarchy,
+)
+
+simpoint = SimPoint(
+    simpoint_list = [3,5,15],
+    weight_list =[0.1,0.5,0.4],
+    simpoint_interval = 1000000,
+    # simpoint_file_path=Path("path/to/simpoints"),
+    # weight_file_path=Path("path/to/weights"),
+)
+
+board.set_se_binary_workload(
+    binary = Resource('x86-print-this'),
+    arguments = ['print this', 15000]
+)
+
+dir = Path("se_checkpoint_folder/cpt.3856914558").as_posix()
+
+def exit():
+    while True:
+        print("MAX_INSTS reached, end of interval")
+        yield True
+
+simulator = Simulator(
+    board=board,
+    checkpoint_path=dir,
+    on_exit_event={
+        ExitEvent.MAX_INSTS: exit()
+    }
+)
+# schedule a MAX_INSTS exit event before the simulation begins
+simulator.schedule_max_insts(simpoint.get_simpoint_interval(), True)
```

Comment:
Can you add a "warmup" period here. I'd like to see how it works to warmup for *x* instructions then run the simpoint for *interval - x* instructions.

Hmm, now that I write that out, maybe it's not quite right. Can you go back to the SimPoint paper and see how they talked about warmup.

Reply:
They mentioned warmup in section 5 and 6. I believe they did the warmup by doing the fast-forwarding: "all
fast forwarding is done so that all of the architecture structures
are completely warmed up when starting simulation (no
cold-start effect)." Later in section 6, the related work section, they mentioned using the entire fast-forward interval for warmup should be the minimum set of fast-forward transactions needed to get to the accurately produce state. They also mentioned checkpoints can either store the architectural state or warm up in a similar manner to the above method.

Reply:
I'm not familiar with warmup but I'll try taking the checkpoint x instructions earlier than the SimPoint and execute the x instructions to perform warmup before taking statistics for the SimPoint. In this case, maybe the number of instructions for warmup should be stored in the SimPoint class too. Please let me know if I misunderstand the warmup process.

Reply:
Yeah, that sounds great. I'm good with adding a warmup instructions parameter to the simpoint object.